pkg/build/api/v1/types.go

```diff
@@ -620,6 +620,9 @@ type GenericWebHookEvent struct {
 
 	// Git is the git information if the Type is BuildSourceGit
 	Git *GitInfo `json:"git,omitempty"`
+
+	// Env contains additional environment variables you want to pass into a builder container
+	Env []kapi.EnvVar `json:"env,omitempty"`
```

Comment:
On Tue, Apr 26, 2016 at 11:57 AM, Jordan Liggitt <notifications@github.com>
wrote:

> In pkg/build/api/v1/types.go
> <https://github.com/openshift/origin/pull/8477#discussion_r61112659>:
>
> > @@ -620,6 +620,9 @@ type GenericWebHookEvent struct {
> >
> >  	// Git is the git information if the Type is BuildSourceGit
> >  	Git *GitInfo `json:"git,omitempty"`
> > +
> > +	// Env contains additional environment variables you want to pass into a builder container
> > +	Env []kapi.EnvVar `json:"env,omitempty"`
>
> is there a way to prevent this if you don't want to give a remote webhook
> power to set envvars in your build?
>
Aside from the on/off switch recently mentioned, were you thinking to go as
far as perhaps a blacklist of requesting IP/ports or something ?

> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8477/files/fde0d585e7272ad1f545600043154b5afd9a9cd0#r61112659>
>